app/packs/controllers/date_validation_controller.js

```diff
@@ -1,71 +1,82 @@
 import { Controller } from "stimulus"
 
 export default class extends Controller {
-  static targets = ["year", "month", "day"]
+  static targets = ["year", "month", "day", "error"]
 
-  change() {
-    this.errors = {}
-    this.mostSignifcantPartsPresent()
-    this.dateInPast()
-    this.displayErrors()
+  connect() {
+    this.validate()
   }
 
-  displayErrors () {
-    if (this.errors.year) {
-      this.yearTarget.classList.add('is-invalid')
-      this.yearTarget.setCustomValidity(this.errors.year)
-    } else {
-      this.yearTarget.classList.remove('is-invalid')
-      this.yearTarget.setCustomValidity('')
-    }
+  change() {
+    this.clear()
+    this.validate()
+  }
 
-    if (this.errors.month) {
-      this.monthTarget.classList.add('is-invalid')
-      this.monthTarget.setCustomValidity(this.errors.month)
-    } else {
-      this.monthTarget.classList.remove('is-invalid')
-      this.monthTarget.setCustomValidity('')
-    }
+  validate() {
+    this.mostSignificantPartsPresent()
+    this.validDateRange()
+  }
 
-    if (this.errors.day) {
-      this.dayTarget.classList.add('is-invalid')
-      this.dayTarget.setCustomValidity(this.errors.day)
-    } else {
-      this.dayTarget.classList.remove('is-invalid')
-      this.dayTarget.setCustomValidity('')
-    }
+  clear() {
+    this.yearTarget.classList.remove('is-invalid')
+    this.yearTarget.setCustomValidity('')
+    this.yearTarget.required = false
+    this.monthTarget.classList.remove('is-invalid')
+    this.monthTarget.setCustomValidity('')
+    this.monthTarget.required = false
+    this.dayTarget.classList.remove('is-invalid')
+    this.dayTarget.setCustomValidity('')
+    this.errorTarget.textContent = ''
   }
 
-  dateInPast() {
```

Comment:
Was dateInPast removed from the front end validations? Why?

Reply:
No, the validation was generalized to do some additional checking about the date range so that function name was updated to reflect that.